api/client/client_test.go

```diff
@@ -748,6 +748,49 @@ func TestClient_WriteTransaction(t *testing.T) {
 			t.Fatalf("failed to delete in write txn: %v", err)
 		}
 	})
+
+	t.Run("test discard write transaction", func(t *testing.T) {
+		id := thread.NewIDV1(thread.Raw, 32)
+		err := client.NewDB(context.Background(), id)
+		checkErr(t, err)
+		err = client.NewCollection(context.Background(), id, db.CollectionConfig{Name: collectionName, Schema: util.SchemaFromSchemaString(schema)})
+		checkErr(t, err)
+
+		existingPerson := createPerson()
+
+		ids, err := client.Create(context.Background(), id, collectionName, Instances{existingPerson})
+		checkErr(t, err)
+
+		existingPerson.ID = ids[0]
+
+		txn, err := client.WriteTransaction(context.Background(), id, collectionName)
+		if err != nil {
+			t.Fatalf("failed to create write txn: %v", err)
+		}
+
+		end, err := txn.Start()
+		if err != nil {
+			t.Fatalf("failed to start write txn: %v", err)
+		}
+		defer func() {
+			err = end()
```

Comment:
@carsonfarmer this is calling the func returned by the Go client that actually just sends a `CloseSend` to the server, which makes it release the transaction... that means it will be committed, but will error if it has been marked as "discarded"

Reply:
You can see how the Go client "ends" the transaction here: https://github.com/textileio/go-threads/blob/master/api/client/write.go#L233 (note the extra error catch...)

Reply:
Yeh this is exactly what I needed. Perfect. One thing I noticed though, in the js client, we're not properly getting the final response as in here: https://github.com/textileio/go-threads/blob/master/api/client/write.go#L237 which we need to ensure the transaction was successful or not.

Reply:
So this has been a useful thread/discussion. Cheers!

Reply:
Yeah totally, I ran into that yesterday: https://github.com/textileio/go-threads/pull/436#discussion_r489016111